NoClassDefFoundError with junit jupiter 5.9

## Overview

- ( ) **Bug report.** The dataprovider is throwing NoClassDefFoundError when using junit jupiter 5.9

With the release of junit 5.9 the ExecutableInvoker class was changed to an Interface. The class itself has been replaced with the DefaultExecutableInvoker class. Therefor the class can't be found any more.

See also release notes for https://junit.org/junit5/docs/current/release-notes/#release-notes-5.9.0

```
org/junit/jupiter/engine/execution/ExecutableInvoker
java.lang.NoClassDefFoundError: org/junit/jupiter/engine/execution/ExecutableInvoker
	at com.tngtech.junit.dataprovider.UseDataProviderInvocationContextProvider.<clinit>(UseDataProviderInvocationContextProvider.java:57)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at org.junit.platform.commons.util.ReflectionUtils.newInstance(ReflectionUtils.java:552)
	at org.junit.platform.commons.util.ReflectionUtils.newInstance(ReflectionUtils.java:527)
	at org.junit.jupiter.engine.extension.MutableExtensionRegistry.registerExtension(MutableExtensionRegistry.java:146)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:274)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at org.junit.jupiter.engine.extension.MutableExtensionRegistry.createRegistryFrom(MutableExtensionRegistry.java:104)
	at org.junit.jupiter.engine.descriptor.ExtensionUtils.populateNewExtensionRegistryFromExtendWithAnnotation(ExtensionUtils.java:74)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.prepare(ClassBasedTestDescriptor.java:149)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.prepare(ClassBasedTestDescriptor.java:84)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$prepare$2(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.prepare(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:90)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:53)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:99)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:79)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:75)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:62)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy5.stop(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$3.run(TestWorker.java:193)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:113)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:65)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
Caused by: java.lang.ClassNotFoundException: org.junit.jupiter.engine.execution.ExecutableInvoker
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 72 more
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -9,9 +9,9 @@ build-job: &build-job
                 # Restart caching for every new wrapper and add job name (= JDK version) as JDK influences Gradle's caching
                 - gradle-repo-v2-{{ .Environment.CIRCLE_JOB }}-{{ checksum "gradle/wrapper/gradle-wrapper.properties" }}
 
-        - run: ./gradlew --build-cache --scan build -Pjunit4Version=4.10  -PjunitJupiterVersion=5.5.0 -PskipSpotBugs
-        - run: ./gradlew --build-cache --scan build -Pjunit4Version=4.11  -PjunitJupiterVersion=5.5.2 -PskipSpotBugs
-        - run: ./gradlew --build-cache --scan build -Pjunit4Version=4.12  -PjunitJupiterVersion=5.6.3 -PskipSpotBugs
+        - run: ./gradlew --build-cache --scan build -Pjunit4Version=4.10  -PjunitJupiterVersion=5.9.0 -PskipSpotBugs
+        - run: ./gradlew --build-cache --scan build -Pjunit4Version=4.11  -PjunitJupiterVersion=5.9.0 -PskipSpotBugs
+        - run: ./gradlew --build-cache --scan build -Pjunit4Version=4.12  -PjunitJupiterVersion=5.9.0 -PskipSpotBugs
         - run: ./gradlew --build-cache --scan build
         - run: COVERALLS_REPO_TOKEN=Npp4tyTSCz0wSMZTJ81vXdVe1uw6WtRrC ./gradlew --build-cache --scan jacocoRootReport coveralls
 

--- a/.travis.yml
+++ b/.travis.yml
@@ -11,9 +11,9 @@ jdk:
   - oraclejdk11
 
 script:
-  - ./gradlew --scan build -Pjunit4Version=4.10  -PjunitJupiterVersion=5.5.0 -PskipSpotBugs
-  - ./gradlew --scan build -Pjunit4Version=4.11  -PjunitJupiterVersion=5.5.2 -PskipSpotBugs
-  - ./gradlew --scan build -Pjunit4Version=4.12  -PjunitJupiterVersion=5.6.3 -PskipSpotBugs
+  - ./gradlew --scan build -Pjunit4Version=4.10  -PjunitJupiterVersion=5.9.0 -PskipSpotBugs
+  - ./gradlew --scan build -Pjunit4Version=4.11  -PjunitJupiterVersion=5.9.0 -PskipSpotBugs
+  - ./gradlew --scan build -Pjunit4Version=4.12  -PjunitJupiterVersion=5.9.0 -PskipSpotBugs
   - ./gradlew --scan build
 
   - cd ${TRAVIS_BUILD_DIR}/junit4/

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -17,7 +17,7 @@ println("Using skipSpotBugs = $skipSpotBugs for current build.")
 // set default junit versions if not set via command line
 val junit4Version by extra(findProperty("junit4Version")?.toString() ?: "4.13.1")
 println("Using JUnit4 version $junit4Version for current build.")
-val junitJupiterVersion by extra(findProperty("junitJupiterVersion")?.toString() ?: "5.7.0")
+val junitJupiterVersion by extra(findProperty("junitJupiterVersion")?.toString() ?: "5.9.0")
 println("Using JUnit Jupiter version $junitJupiterVersion for current build.")
 
 class Dependency {

--- a/junit-jupiter-params/pom.xml
+++ b/junit-jupiter-params/pom.xml
@@ -70,12 +70,12 @@
         <dependency>
             <groupId>org.junit.jupiter</groupId>
             <artifactId>junit-jupiter-params</artifactId>
-            <version>5.5.0</version>
+            <version>5.9.0</version>
         </dependency>
         <dependency>
             <groupId>org.junit.jupiter</groupId>
             <artifactId>junit-jupiter-engine</artifactId>
-            <version>5.5.0</version>
+            <version>5.9.0</version>
         </dependency>
         <dependency>
             <groupId>com.github.spotbugs</groupId>

--- a/junit-jupiter-params/src/main/java/com/tngtech/junit/dataprovider/AbstractUseDataProviderArgumentProvider.java
+++ b/junit-jupiter-params/src/main/java/com/tngtech/junit/dataprovider/AbstractUseDataProviderArgumentProvider.java
@@ -173,6 +173,7 @@ public abstract class AbstractUseDataProviderArgumentProvider<SOURCE_ANNOTATION 
 
     private ConfigurationParameters emptyConfigurationParameters() {
         return new ConfigurationParameters() {
+            @SuppressWarnings("deprecation")
             @Override
             public int size() {
                 return 0;

--- a/junit-jupiter-params/src/main/java/com/tngtech/junit/dataprovider/AbstractUseDataProviderArgumentProvider.java
+++ b/junit-jupiter-params/src/main/java/com/tngtech/junit/dataprovider/AbstractUseDataProviderArgumentProvider.java
@@ -30,10 +30,9 @@ import org.junit.jupiter.api.TestInfo;
 import org.junit.jupiter.api.extension.ExtensionContext;
 import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
 import org.junit.jupiter.api.extension.ExtensionContext.Store;
-import org.junit.jupiter.api.extension.InvocationInterceptor;
 import org.junit.jupiter.api.extension.ParameterResolutionException;
 import org.junit.jupiter.engine.config.DefaultJupiterConfiguration;
-import org.junit.jupiter.engine.execution.InterceptingExecutableInvoker;
+import org.junit.jupiter.engine.execution.DefaultExecutableInvoker;
 import org.junit.jupiter.engine.extension.ExtensionRegistry;
 import org.junit.jupiter.params.provider.Arguments;
 import org.junit.platform.engine.ConfigurationParameters;
@@ -53,8 +52,6 @@ import com.tngtech.junit.dataprovider.resolver.DataProviderResolverContext;
  */
 public abstract class AbstractUseDataProviderArgumentProvider<SOURCE_ANNOTATION extends Annotation, DATAPROVIDER_ANNOTATION extends Annotation>
         extends AbstractDataProviderArgumentProvider<SOURCE_ANNOTATION> {
-
-    private static final InterceptingExecutableInvoker executableInvoker = new InterceptingExecutableInvoker();
 
     protected static final Namespace NAMESPACE_USE_DATAPROVIDER = Namespace
             .create(AbstractUseDataProviderArgumentProvider.class, "dataCache");
@@ -156,8 +153,7 @@ public abstract class AbstractUseDataProviderArgumentProvider<SOURCE_ANNOTATION 
             // TODO how to not require junit-jupiter-engine dependency and reuse already existing ExtensionRegistry?
             ExtensionRegistry extensionRegistry = createRegistryWithDefaultExtensions(
                     new DefaultJupiterConfiguration(emptyConfigurationParameters()));
-            Object data = executableInvoker.invoke(dataProviderMethod, context.getTestInstance().orElse(null), context,
-                    extensionRegistry, InvocationInterceptor::interceptTestFactoryMethod);
+            Object data = new DefaultExecutableInvoker(context, extensionRegistry).invoke(dataProviderMethod, context.getTestInstance().orElse(null));
             if (cacheDataProviderResult) {
                 store.put(dataProviderMethod, data);
             }

--- a/junit-jupiter-params/src/main/java/com/tngtech/junit/dataprovider/AbstractUseDataProviderArgumentProvider.java
+++ b/junit-jupiter-params/src/main/java/com/tngtech/junit/dataprovider/AbstractUseDataProviderArgumentProvider.java
@@ -21,7 +21,9 @@ import static org.junit.jupiter.engine.extension.MutableExtensionRegistry.create
 
 import java.lang.annotation.Annotation;
 import java.lang.reflect.Method;
+import java.util.Collections;
 import java.util.Optional;
+import java.util.Set;
 import java.util.stream.Stream;
 
 import org.junit.jupiter.api.TestInfo;
@@ -31,7 +33,7 @@ import org.junit.jupiter.api.extension.ExtensionContext.Store;
 import org.junit.jupiter.api.extension.InvocationInterceptor;
 import org.junit.jupiter.api.extension.ParameterResolutionException;
 import org.junit.jupiter.engine.config.DefaultJupiterConfiguration;
-import org.junit.jupiter.engine.execution.ExecutableInvoker;
+import org.junit.jupiter.engine.execution.InterceptingExecutableInvoker;
 import org.junit.jupiter.engine.extension.ExtensionRegistry;
 import org.junit.jupiter.params.provider.Arguments;
 import org.junit.platform.engine.ConfigurationParameters;
@@ -52,7 +54,7 @@ import com.tngtech.junit.dataprovider.resolver.DataProviderResolverContext;
 public abstract class AbstractUseDataProviderArgumentProvider<SOURCE_ANNOTATION extends Annotation, DATAPROVIDER_ANNOTATION extends Annotation>
         extends AbstractDataProviderArgumentProvider<SOURCE_ANNOTATION> {
 
-    private static final ExecutableInvoker executableInvoker = new ExecutableInvoker();
+    private static final InterceptingExecutableInvoker executableInvoker = new InterceptingExecutableInvoker();
 
     protected static final Namespace NAMESPACE_USE_DATAPROVIDER = Namespace
             .create(AbstractUseDataProviderArgumentProvider.class, "dataCache");
@@ -184,6 +186,11 @@ public abstract class AbstractUseDataProviderArgumentProvider<SOURCE_ANNOTATION 
             @Override
             public Optional<String> get(String key) {
                 return Optional.empty();
+            }
+
+            @Override
+            public Set<String> keySet() {
+                return Collections.emptySet();
             }
         };
     }

--- a/junit-jupiter/pom.xml
+++ b/junit-jupiter/pom.xml
@@ -73,7 +73,7 @@
         <dependency>
             <groupId>org.junit.jupiter</groupId>
             <artifactId>junit-jupiter-engine</artifactId>
-            <version>5.5.0</version>
+            <version>5.9.0</version>
         </dependency>
         <dependency>
             <groupId>com.github.spotbugs</groupId>

--- a/junit-jupiter/src/main/java/com/tngtech/junit/dataprovider/UseDataProviderInvocationContextProvider.java
+++ b/junit-jupiter/src/main/java/com/tngtech/junit/dataprovider/UseDataProviderInvocationContextProvider.java
@@ -33,7 +33,7 @@ import org.junit.jupiter.api.extension.*;
 import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
 import org.junit.jupiter.api.extension.ExtensionContext.Store;
 import org.junit.jupiter.engine.config.DefaultJupiterConfiguration;
-import org.junit.jupiter.engine.execution.InterceptingExecutableInvoker;
+import org.junit.jupiter.engine.execution.DefaultExecutableInvoker;
 import org.junit.jupiter.engine.extension.ExtensionRegistry;
 import org.junit.platform.engine.ConfigurationParameters;
 
@@ -52,8 +52,6 @@ import com.tngtech.junit.dataprovider.resolver.DataProviderResolverContext;
  */
 public abstract class UseDataProviderInvocationContextProvider<TEST_ANNOTATION extends Annotation, DATAPROVIDER_ANNOTATION extends Annotation>
         extends AbstractDataProviderInvocationContextProvider<TEST_ANNOTATION> {
-
-    private static final InterceptingExecutableInvoker executableInvoker = new InterceptingExecutableInvoker();
 
     protected static final Namespace NAMESPACE_USE_DATAPROVIDER = Namespace
             .create(UseDataProviderInvocationContextProvider.class, "dataCache");
@@ -175,8 +173,7 @@ public abstract class UseDataProviderInvocationContextProvider<TEST_ANNOTATION e
             // TODO how to not require junit-jupiter-engine dependency and reuse already existing ExtensionRegistry?
             ExtensionRegistry extensionRegistry = createRegistryWithDefaultExtensions(
                     new DefaultJupiterConfiguration(emptyConfigurationParameters()));
-            Object data = executableInvoker.invoke(dataProviderMethod, context.getTestInstance().orElse(null), context,
-                    extensionRegistry, InvocationInterceptor::interceptTestFactoryMethod);
+            Object data = new DefaultExecutableInvoker(context, extensionRegistry).invoke(dataProviderMethod, context.getTestInstance().orElse(null));
             if (cacheDataProviderResult) {
                 store.put(dataProviderMethod, data);
             }

--- a/junit-jupiter/src/main/java/com/tngtech/junit/dataprovider/UseDataProviderInvocationContextProvider.java
+++ b/junit-jupiter/src/main/java/com/tngtech/junit/dataprovider/UseDataProviderInvocationContextProvider.java
@@ -22,19 +22,18 @@ import static org.junit.jupiter.engine.extension.MutableExtensionRegistry.create
 
 import java.lang.annotation.Annotation;
 import java.lang.reflect.Method;
+import java.util.Collections;
 import java.util.List;
 import java.util.Optional;
+import java.util.Set;
 import java.util.stream.Stream;
 
 import org.junit.jupiter.api.TestInfo;
-import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.*;
 import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
 import org.junit.jupiter.api.extension.ExtensionContext.Store;
-import org.junit.jupiter.api.extension.InvocationInterceptor;
-import org.junit.jupiter.api.extension.ParameterResolutionException;
-import org.junit.jupiter.api.extension.TestTemplateInvocationContext;
 import org.junit.jupiter.engine.config.DefaultJupiterConfiguration;
-import org.junit.jupiter.engine.execution.ExecutableInvoker;
+import org.junit.jupiter.engine.execution.InterceptingExecutableInvoker;
 import org.junit.jupiter.engine.extension.ExtensionRegistry;
 import org.junit.platform.engine.ConfigurationParameters;
 
@@ -54,7 +53,7 @@ import com.tngtech.junit.dataprovider.resolver.DataProviderResolverContext;
 public abstract class UseDataProviderInvocationContextProvider<TEST_ANNOTATION extends Annotation, DATAPROVIDER_ANNOTATION extends Annotation>
         extends AbstractDataProviderInvocationContextProvider<TEST_ANNOTATION> {
 
-    private static final ExecutableInvoker executableInvoker = new ExecutableInvoker();
+    private static final InterceptingExecutableInvoker executableInvoker = new InterceptingExecutableInvoker();
 
     protected static final Namespace NAMESPACE_USE_DATAPROVIDER = Namespace
             .create(UseDataProviderInvocationContextProvider.class, "dataCache");
@@ -206,6 +205,11 @@ public abstract class UseDataProviderInvocationContextProvider<TEST_ANNOTATION e
             @Override
             public Optional<String> get(String key) {
                 return Optional.empty();
+            }
+
+            @Override
+            public Set<String> keySet() {
+                return Collections.emptySet();
             }
         };
     }

--- a/junit-jupiter/src/main/java/com/tngtech/junit/dataprovider/UseDataProviderInvocationContextProvider.java
+++ b/junit-jupiter/src/main/java/com/tngtech/junit/dataprovider/UseDataProviderInvocationContextProvider.java
@@ -29,9 +29,11 @@ import java.util.Set;
 import java.util.stream.Stream;
 
 import org.junit.jupiter.api.TestInfo;
-import org.junit.jupiter.api.extension.*;
+import org.junit.jupiter.api.extension.ExtensionContext;
 import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
 import org.junit.jupiter.api.extension.ExtensionContext.Store;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.TestTemplateInvocationContext;
 import org.junit.jupiter.engine.config.DefaultJupiterConfiguration;
 import org.junit.jupiter.engine.execution.DefaultExecutableInvoker;
 import org.junit.jupiter.engine.extension.ExtensionRegistry;

--- a/junit-jupiter/src/main/java/com/tngtech/junit/dataprovider/UseDataProviderInvocationContextProvider.java
+++ b/junit-jupiter/src/main/java/com/tngtech/junit/dataprovider/UseDataProviderInvocationContextProvider.java
@@ -192,6 +192,7 @@ public abstract class UseDataProviderInvocationContextProvider<TEST_ANNOTATION e
 
     private ConfigurationParameters emptyConfigurationParameters() {
         return new ConfigurationParameters() {
+            @SuppressWarnings("deprecation")
             @Override
             public int size() {
                 return 0;

